src/OrchardCore/OrchardCore.Workflows.Abstractions/Services/IWorkflowManager.cs

```diff
@@ -37,7 +37,7 @@ public interface IWorkflowManager
         /// <param name="isAlwaysCorrelated">
         /// If true, to be correlated a workflow instance only needs to be halted on an event activity of the related type, regardless the 'correlationId'. False by default.
         /// </param>
-        Task TriggerEventAsync(string name, IDictionary<string, object> input = null, string correlationId = null, bool isExclusive = false, bool isAlwaysCorrelated = false);
+        Task<IEnumerable<WorkflowExecutionContext>> TriggerEventAsync(string name, IDictionary<string, object> input = null, string correlationId = null, bool isExclusive = false, bool isAlwaysCorrelated = false);
```

Comment:
This will introduce a breaking change. Can you think of a way to not bring a breaking change?

Reply:
Yes, I initially tried this without modifying the OC source code, but found that there might be a problem here, because if I implemented a dictionary again somewhere else, it would lose its limiting effect
![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/4d181104-92cf-4cb2-b412-c766b5501df7)


But the good news is that the changes here should have little impact, like here

![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/4f7b0fb6-eb5b-4fcf-af0d-9e2ae8a042c2)

Reply:
My side is fine, but I still hope to pass this PR

![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/9159bb6b-e35c-4e08-9028-9fbe1fbe0f22)
![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/465cc1bd-0b9b-4396-ae9e-3c2aa731fb93)

Reply:
Yes I understand. I am just trying to avoid the breaking change. @jtkech how do you feel about introducing a breaking change here?

Alternatively, we can add this to the interface which will not introduce a breaking change. But I don't like this approach

```
        async Task<IEnumerable<WorkflowExecutionContext>> TriggerAndGetEventAsync(string name, IDictionary<string, object> input = null, string correlationId = null, bool isExclusive = false, bool isAlwaysCorrelated = false)
        {
            await TriggerEventAsync(name, input, correlationId, isExclusive, isAlwaysCorrelated);

            return Enumerable.Empty<WorkflowExecutionContext>();
        }
```

then we can obsolete the `TriggerEventAsync` method. Finally, implement the new `TriggerAndGetEventAsync` in the `WorkflowManager` so this way anyone that is using `IWorkflowManager` will not encounter the breaking change. but anyone that is using our implementation will get the correct result.